cosmos/operators/_asynchronous/base.py

```diff
@@ -34,7 +39,70 @@ def _create_async_operator_class(profile_type: str, dbt_class: str) -> Any:
         raise ImportError(f"Error in loading class: {class_path}. Unable to find the specified operator class.") from e
 
 
+def configure_datasets(operator_kwargs: dict[str, Any], task_id: str) -> dict[str, Any]:
+    """
+    Sets the outlets for the operator by creating a DatasetAlias.
+
+    :param operator_kwargs: (dict[str, Any])
+    :param task_id: (str)
+    :return: (dict) Updated operator_kwargs
+    """
+    from airflow.datasets import DatasetAlias
+
+    # ignoring the type because older versions of Airflow raise the follow error in mypy
+    # error: Incompatible types in assignment (expression has type "list[DatasetAlias]", target has type "str")
+    dag_id = operator_kwargs.get("dag")
+    task_group_id = operator_kwargs.get("task_group")
+    operator_kwargs["outlets"] = [
+        DatasetAlias(name=get_dataset_alias_name(dag_id, task_group_id, task_id))
+    ]  # type: ignore
+
+    return operator_kwargs
+
+
+def get_remote_sql(async_context: dict[str, Any], project_dir: str) -> str:
```

Comment:
It might make sense to move this method to the `io` module https://github.com/astronomer/astronomer-cosmos/blob/main/cosmos/io.py and re-use everywhere (refactoring in BigQuery can be a separate PR), but for now, we could at least add this method there